tools/cellprofiler/cellprofiler.xml

```diff
@@ -0,0 +1,108 @@
+<tool id="cp319" name="Cellprofiler" version="@CP_VERSION@">
+    <description>Run a CellProfiler pipeline</description>
+    
+    <macros>
+        <import>macros.xml</import>
+        <xml name="test_assert_content" token_n="291">
+            <assert_contents>
+                <has_n_lines n="@N@" />
+            </assert_contents>
+        </xml>
+    </macros>
+
+    <expand macro="requirements" version="@CP_VERSION@" package="cellprofiler" />
+    <expand macro="stdio" />
+
+    <command><![CDATA[
+        bash '$script_file' &&
+        cellprofiler -c -r --file-list list.txt -o ./output -p '$pipeline' 
+    ]]></command>
+
+    <configfiles>
+        <configfile name="script_file">
+mkdir ./input
+mkdir ./output
+#for $i,$filename in enumerate($input_type.inputdir):
+    ln -s '$filename' ./input/${filename.element_identifier}.${filename.ext}
+#end for
+
+find `pwd`/input -name *.tiff > list.txt
```

Comment:
shouldn't we have more file types here?

Reply:
Cell Profiler says [here](http://cellprofiler-manual.s3.amazonaws.com/CellProfiler-3.0.0/modules/input.html#:~:text=CellProfiler%20can%20read%20a%20wide,here%20for%20the%20formats%20available.&text=For%20image%20analysis%20purposes%2C%20a,TIF%20or%20PNG%20is%20recommended.): 

> CellProfiler can read a wide variety of image formats by using a library called Bio-Formats; see [here](https://docs.openmicroscopy.org/bio-formats/5.9.2/supported-formats.html) for the formats available.

Reply:
@bgruening can we rely on the label ["galaxy.datatypes.images"](https://github.com/galaxyproject/galaxy/blob/66db1c5c62c6ad12d5ffea6f312b50a44dac95fb/lib/galaxy/config/sample/datatypes_conf.xml.sample#L932) to filter the ones supported by Cell Profiler?

Reply:
Yes, these are the supported file types. We could add more but for the beginning, I would start with the supported once.

Reply:
Then I guess the list of file extensions that both Galaxy and Cell Profiler support is this:

`.jpg
.png
.tif
.tiff
.bmp
.gif
.pcx
.ppm
.psd
.pbm
.pgm
.eps
`